1-static-analysis.md

```diff
@@ -0,0 +1,298 @@
+# Surface Module Source Static Analysis
+
+## Synopsis
+
+Extend instances of `ModuleSource` such that they reflect certain results of
+static analysis, like their `import` and `export` bindings, such that tools can
+inspect module graphs.
+
+## Dependencies
+
+This proposal depends on [Module and ModuleSource][0] from the [Compartments
+proposal](README.md) to introduce `ModuleSource`.
+
+## Design
+
+Extend [ModuleSource][0], such that instances have the following properties:
+
+- `bindings`, an `Array` of `Binding`s.
+- `needsImportMeta`, a `boolean` indicating that the module contains
+  `import.meta` syntax.
+
+Where a `Binding` is an ordinary `Object` with one of the valid binding shapes
+for each name or wildcard (`*`) bound by `import` or `export` in the text
+of the module, in their order of appearance.
+
+- `{ import: string, from: string }`
+
+  For example, `import { a } from 'a.js'` would produce
+  `{ import: 'a', from: 'a.js' }`.
+
+  For example, `import { a, b } from 'ab.js'` would produce:
+  `{ import: 'a', from: 'ab.js' }` and
+  `{ import: 'b', from: 'ab.js' }`.
+
+- `{ import: string, as: string, from: string }`
+
+  For example, `import { a as x } from 'a.js'` would produce
```

Comment:
I'm having a hard time understanding why is `x` relevant here, since `x` is internal to the module source. Basically, in order for a tool to replace this import statement, they will need an AST and make transformations, while this is just meta that might help with this transformation but we should probably stay away from exposing internals of the source.